### PR TITLE
STORM-3786 ensure v2 metrics tick reporting occurs at specified interval

### PR DESCRIPTION
**Pull request by agresch**
## What is the purpose of the change

Prevent under/over reporting of v2 metrics tick.

## How was the change tested

Ran storm-client unit tests.  Validated LoggingMetricsConsumer logged v2 metrics on worker.

### CODE REVIEW

File: storm-client/src/jvm/org/apache/storm/executor/Executor.java [hunk 1]
```diff
@@ -462,6 +473,30 @@ protected void setupMetrics() {
...
                 }
             );
+            if (interval == v2MetricsTickInterval) {
+                v2TickScheduled = true;
+            }
+        }
+
+        if (!v2TickScheduled) {
+            LOG.info("Scheduling v2 metrics tick for interval {}", v2MetricsTickInterval);
+            StormTimer timerTask = workerData.getUserTimer();
```
Comment (Ethanlm) [inline storm-client/src/jvm/org/apache/storm/executor/Executor.java:483]: This code looks similar to above. Can we de-duplicate?

**Review by Ethanlm — COMMENTED** [review]
Overall looks good to me

File: storm-client/src/jvm/org/apache/storm/executor/Executor.java (hunk 1 above)
Comment (agresch) [reply storm-client/src/jvm/org/apache/storm/executor/Executor.java:483]: sure, was debating same thing as I committed.

Comment (agresch) [reply storm-client/src/jvm/org/apache/storm/executor/Executor.java]: updated

**Review by Ethanlm — APPROVED** [review]
+1

**Pull request merged**